lib/child_process.js

```diff
@@ -613,8 +595,34 @@ function normalizeSpawnArguments(file, args, options) {
 
 function spawn(file, args, options) {
   const child = new ChildProcess();
-
   options = normalizeSpawnArguments(file, args, options);
+
+  if (options.signal) {
+    const signal = options.signal;
+    // Validate signal, if present
+    validateAbortSignal(signal, 'options.signal');
+
+    // Do nothing and throw if already aborted
+    if (signal.aborted) {
+      onAbortListener();
+    } else {
+      signal.addEventListener('abort', onAbortListener, { once: true });
+      child.once('close',
+                 () =>
+                   signal
+                    .removeEventListener('abort', onAbortListener)
+      );
+    }
+
+    function onAbortListener() {
+      process.nextTick(() => {
+        child?.kill?.(options.killSignal);
+
+        child.emit('error', new AbortError());
+      });
+    }
```

Comment:
Maybe move this into the `if (options.signal)` block as this is the only place it's being used.

Reply:
Maybe I'm missing something, but it is within the `if (options.signal)` block, or what do you mean exactly?

Reply:
You're right, I miscount the number of brackets, my bad 😅